src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -237,6 +244,14 @@ export default function (props: ConnectDevicesMenuProps) {
             healthConnectStatus={healthConnectStatus} />;
     }
 
+    function getEnvironmentalMenuItem() {
+        if (!accountTypes.includes("Environmental") || !props.postalCodeSurveyName) {
```

Comment:
This should work fine, but the second condition could also be moved up to the account type filtering so you can just check the account types here like we do for the other menu items.

```
if ((!settings?.airQualityEnabled && !settings?.weatherEnabled) || !props.postalCodeSurveyName) { 
	accountTypes = accountTypes.filter(a => a != "Environmental");
}
```